### PR TITLE
change seed, otherwise random number will not be changed

### PR DESCRIPTION
**Pull request by SRC-xiaojin**
(cherry picked from commit aa6f8bbb5512ae5ea542a0ebf7e808778e26f377)

## What is the purpose of the change

change seed, otherwise random number will not be changed

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when a cross-module dependency exists.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

File: src/protocol/TopicRouteData.h [hunk 1]
```diff
@@ -122,11 +122,13 @@ class TopicRouteData {
   std::string selectBrokerAddr() {
     int bdSize = m_brokerDatas.size();
     if (bdSize > 0) {
+      std::srand(std::time(0) + bdSize);
```
Comment (ifplusor) [inline src/protocol/TopicRouteData.h:125]: Don't set seed every time.

**Review by ifplusor — CHANGES_REQUESTED** [review]

File: src/protocol/TopicRouteData.h (hunk 1 above)
Comment (ifplusor) [reply src/protocol/TopicRouteData.h:125]: I think setting seed in an init function somewhere is good.

Comment (ShannonDing) [reply src/protocol/TopicRouteData.h:125]: > 



> I think setting seed in an init function somewhere is good.

agree

File: src/MQClientManager.cpp [hunk 2]
```diff
@@ -19,7 +19,9 @@
 
 namespace rocketmq {
 //<!************************************************************************
-MQClientManager::MQClientManager() {}
+MQClientManager::MQClientManager() {
+  std::srand(std::time(0));
```
Comment (ifplusor) [inline src/MQClientManager.cpp:23]: Please include `cstdlib` and `ctime`.

**Review by ifplusor — CHANGES_REQUESTED** [review]

**Review by ifplusor — APPROVED** [review]

**Pull request merged**